apps/site/lib/site_web/controllers/schedule/hours_of_operation.ex

```diff
@@ -10,7 +10,7 @@ defmodule SiteWeb.ScheduleController.HoursOfOperation do
   def call(%Plug.Conn{assigns: %{route: route}} = conn, _opts) when not is_nil(route) do
     route.id
     |> full_route_id
-    |> Schedules.Repo.hours_of_operation(conn.assigns.date)
+    |> Schedules.Repo.hours_of_operation(conn.assigns.date, route.description)
     |> assign_hours(conn)
```

Comment:
My original intent was to remove the plug.  I was hoping to get all of the hours from the endpoint.  I kept it around because I thought it was out of scope to update the way the default hours were handled.  Keeping this plug around allows the hours to be used on the default page correctly.

You are right in thinking the way it currently is written is slightly less performant as we have to send the same data twice over the network (once for page load, and again on component load).

I see 2 paths forward.

- Keep as is and update the non-rapid transit hours further down the line to use the end-point.
- Change the RapidTransit page to not use the endpoint but use the data embedded into the page instead.

I personally don't like how the data is embedded into the page but that's my personal bias and I can't see anything wrong with it outside of not being able to fetch new data dynamically if need be.

Reply:
Totally agree with your assessment on keeping the plug in for now. Thanks for sharing your thoughts here.

:sparkles: For the record I agree embedding data into `<script>` tags isn't so great to work with, and over time we might converge on a more modern way. 

As you imply, interacting with a specialized endpoint from the frontend definitely improves our situation when it comes to customizing the fetch and dynamically updating UI. I also want us to be mindful it's not the **only** approach; not every UI problem requires customization or dynamic updating like that, and sometimes using data to directly render HTML (.eex templates) can work just as well as the data -> JS network request -> JSX template workflow.

My own opinion is that this specific task _could_ have been handled with the static data from the existing plug written into a template, but I admit this particular page is a total mixed bag when it comes to component interactions (there are some! sorta) and degree of "dynamicness" expected. Not penalizing your approach specifically, but just wanted to offer some perspectives -- thanks for letting me share these thoughts!